### ISSUE TITLE
Evaluate on command?

### ISSUE THREAD

**Issue by EricR** [victorb/obsidian-wielder#2]
Is it possible to evaluate code blocks on command rather than instantaneously as code gets typed? I could see this potentially being problematic if I’m typing code that does something state changing like, say, modifies my vault (e.g., I’m deleting a file by path name and I’m typing out the path name in quotes).

**Comment by victorb** [comment]
Yeah, could provide this as an setting for the plugin, something like `Evaluate automatically` which when toggled off, would require the user to run a command to evaluate the document.

Although, I'd advise against the approach of doing potentially destructive things without wrapping them in something that requires user action.

So instead of just doing something like this, which would delete a file when evaluated (so when the document is loaded, by default):

```clojure
(delete-file filename)
```

You'd do something like this:

```clojure
(defn $button [filename]
  [:button
    {:onClick #(delete-file filename)}
    "Delete " filename])

(*renderReagent [$button "file-to-be-deleted"])
```

Which would render a button to be clicked to have the file to be deleted.

With that said, having the option of choosing when evaluation happens is not a bad idea :)

**Comment by matthewdowney** [comment]
Just chiming in to say that I'd be in favor of a mode that requires explicit action on behalf of the user to execute code
